src/litdata/streaming/item_loader.py

```diff
@@ -126,6 +126,7 @@ def deserialize(self, raw_item_data: bytes) -> "PyTree":
         for size, data_format in zip(sizes, self._config["data_format"]):
             serializer = self._serializers[self._data_format_to_key(data_format)]
             data_bytes = raw_item_data[idx : idx + size]
+            serializer.setup(data_format)
```

Comment:
We should probably do this only once to avoid losing time. Also, would you mind adding a test ?

Reply:
Yes, single call makes sense. However, single call would require quite a refactoring of the code. I suggest this as a temporary fix. I will write a test.

Reply:
It could be done on instantiation in the base class there: https://github.com/Lightning-AI/litdata/pull/96/files/49976880c0279a075697ec3bc6b47b8d7549c51d#diff-378fc82b9b3cccccce134a1610f9dd2db00ee3d87b361ad84617687fef081790R39